message-index/messages/GHC-48361/index.md

```diff
@@ -0,0 +1,10 @@
+---
+title: Binding type variables is not allowed in pattern bindings
+summary: You can only bind value-level variables in a pattern, not type variables.
+severity: error
+introduced: 9.6.1
+---
+
+Patterns bind values to variables.  It would be interesting to allow
+for types variables, which you could then use in type annotations on
+the right-hand side.  But this is not supported by this ghc version.
```

Comment:
```suggestion
Patterns can bind constructor arguments to variables, but they are not able to bind type arguments to type variables.
```
I feel like this description as submitted is more addressed to people who know a lot about compilers and functional programming.